apps/prairielearn/src/lib/question-submission.ts

```diff
@@ -99,6 +100,45 @@ export async function processSubmission(
     await saveSubmission(submission, variant, res.locals.question, res.locals.course);
     return submission.variant_id;
   } else {
-    throw new error.HttpStatusError(400, `unknown __action: ${req.body.__action}`);
+    throw new HttpStatusError(400, `unknown __action: ${req.body.__action}`);
   }
 }
+
+export async function reportIssueFromForm(
```

Comment:
> Why is this in `lib/question-submission.ts`? It feels like this belongs in maybe `lib/issues.ts`?

I originally added it to `lib/issues.ts`, but this [created a circular dependency](https://github.com/PrairieLearn/PrairieLearn/actions/runs/9175738250/job/25229370193). I moved it to question-submission because this function already contains code that handles the submission process from a question, which is similar across multiple contexts.

As I review that other function, it uses a different signature (a boolean instead of an enum to determine the context, and in a different order of arguments). I can change this function to use the same kind of signature.

Reply:
Reordered in ae9bd62.

Reply:
It seems like you could break the loop by moving `validateVariantAgainstQuestion` somewhere more appropriate, perhaps `models/variant.ts`? The fact that that function is now being used on variant insertion means that `lib/question-submission.ts` probably isn't the right place for it.

Reply:
Done in fff6ed405563fa448e720c2d762cfe6153529037 and 012617429a8e78a84c4e8f660294a5b773ffda7f.